test/integration/controller/core/suite_test.go

```diff
@@ -84,6 +85,8 @@ func managerSetup(mgr manager.Manager, ctx context.Context) {
 	cCache := cache.New(mgr.GetClient())
 	queues := queue.NewManager(mgr.GetClient(), cCache)
 
+	gomega.Expect(features.SetEnable(features.ClusterQueueVisibility, true)).To(gomega.Succeed())
```

Comment:
If we enabled featureGate when setting up the manager, we can not test the behavior when disabling this featureGate. (a) Could you enable the featureGate only when testing this feature? (b) Or could you expand test cases so that we can verify behavior in the same cases for the cq-controller when enabling and disabling this?

I think (a) might be better since (b) might be overkill. Maybe, we can take (b) in unit tests if needed.

Reply:
+1. I would suggest following the pattern as for `features.PartialAdmission` (setting the feature gate for only `It`(s) that test the feature. I think we can move more testing to unit tests.

Reply:
I'm not sure that we can use here the same pattern as for `features.PartialAdmission`. As long as we use the start function to create a snapshot where the feature flag is checked, we must be sure that this will be declared before the core.SetupControllers are initialized. Maybe, it can be changed #1099

Reply:
I see, good point. In that case we might need to move the contents of `BeforeSuite` from `suite_test.go` to `BeforeAll` in per describe in the test files, such as `clusterqueue_controller_test.go`. Similarly as we do for `jobs/job/job_controller_test.go`

Reply:
added beforeAll instead of BeforeSuite https://github.com/kubernetes-sigs/kueue/pull/1139/commits/b92342ff7e039eed2206ff656926e42a29107a02